### PR TITLE
Geo address

### PR DESCRIPTION
**Pull request by defong**
### Context
Geo address to get a the latitude and longitude of an address.

### Changes proposed in this pull request
Added addition fields to `Location` table called `GeoAddress`

### Guidance to review
States.
1) The existing Location data 'Address' will be cloned into `GeoAddress`, via migrations.
2) When a course is publish, existing Campus, will use a new `Location` data based on existing old `Location`'s latitude and longitude, existing `Location`'s `Address` will basically be the same as a `Campus` Address minus the `Campus`'s name.
3) Therefore the display of a 'Campus' will exhibit the same inaccurate as beforehand.
4) When the geo coding kicks in then the `Campus`'s `Location`'s latitude and longitude will be updated.
5) The display of a 'Campus' will be more accurate 
6) `Campus` with name `Main site` or without name or without address will be using `Location`'s address without `Campus`'s Name as prefix
 

Paired with @MartinBentley

### CODE REVIEW

**Comment by defong** [comment]
[Use training location name when we geocode training locations](https://trello.com/c/Tcalw3yq/732-use-training-location-name-when-we-geocode-training-locations)

File: src/geocoder/LocationRequester.cs [hunk 1]
```diff
@@ -40,9 +40,14 @@ public async Task RequestLocations()
 
             foreach (var location in locations)
             {
+                if(string.IsNullOrWhiteSpace(location.GeoAddress))
+                {
+                    location.GeoAddress = location.Address;
```
Comment (MartinBentley) [inline src/geocoder/LocationRequester.cs]: Is this right?

**Comment by fofr** [comment]
Are there some before and after example lat/longs as a result of this?

**Comment by defong** [comment]
@fofr 


image

**Comment by dankmitchell** [comment]
Great work @fofr 💪

**Pull request merged**